src/distribute_kv_cache/rpc/utils.rs

```diff
@@ -0,0 +1,123 @@
+use bytes::{Buf, BytesMut};
+use tracing::debug;
+
+use super::error::RpcError;
+
+/// Read with timeout options from the environment variables
+#[macro_export]
+macro_rules! read_exact_timeout {
+    ($self:expr, $dst:expr, $read_timeout:expr) => {
+        async move {
+            use tokio::time::timeout;
+
+            match timeout($read_timeout, $self.read_exact($dst)).await {
+                Ok(res) => match res {
+                    Ok(size) => Ok(size),
+                    Err(read_err) => Err(read_err),
+                },
+                Err(timeout_err) => Err(timeout_err.into()),
+            }
+        }
+    };
+}
+
+/// Write with timeout options from the environment variables
+#[macro_export]
+macro_rules! write_all_timeout {
+    ($self:expr, $src:expr, $write_timeout:expr) => {
```

Comment:
please name it `writer` instead of `self`

Reply:
Got.